openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+
+while [ -n "$1" ]; do
+	case "$1" in
+		--version|-v) export VERSION=1; break;;
+		--url) export BASE_URL="$2"; shift;;
+		--uuid) export UUID="$2"; shift;;
+		--key) export KEY="$2"; shift;;
+		--verify_ssl) export verify_ssl="$2"; shift;;
+		--interval) export INTERVAL="$2"; shift;;
+		--monitored_interfaces) export MONITORED_INTERFACES=$2; shift;;
+		-*)
+			echo "Invalid option: $1"
+			exit 1
+		;;
+		*) break;
+	esac
+	shift;
+done
+
+if [ "$VERSION" -eq "1" ]; then
+	VERSION=$(cat /etc/monitoring/VERSION)
+	echo "openwisp-monitoring $VERSION"
+	exit 0
+fi
+
+if [ "$verify_ssl" = 0 ]; then
+    curl_command='curl -k'
+else
+    curl_command='curl'
+fi
+
+url="$BASE_URL/api/v1/monitoring/device/$UUID/?key=$KEY"
```

Comment:
`url` > `URL`

make sure any variable containing constants like this (variables that do not change) are uppercase please.